syncsession.go

```diff
@@ -13,23 +13,35 @@ type SyncSession struct {
 	logger             logger.Logger
 	context            *SyncContext
 	authenticatioToken string
+	sessionKey   string
 }
 
 func newSyncSession(parentLogger logger.Logger,
 	context *SyncContext,
 	username string,
 	password string,
-	label string) (*SyncSession, error) {
+	label string,
+	sessionKey string) (*SyncSession, error) {
+
+	if sessionKey != "" {
+		return &SyncSession{
+			logger:             parentLogger.GetChild("session"),
+			context:            context,
+			sessionKey:			sessionKey,
+		}, nil
+
+	} else {
```

Comment:
You return unconditionally in the `if`, no need for else.
Also, please gofmt your code (see indentation issue)